homeassistant/components/slack/number.py

```diff
@@ -0,0 +1,45 @@
+"""Slack platform for select component."""
+
+from __future__ import annotations
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SlackEntity
+from .const import DATA_CLIENT, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Slack select."""
+    async_add_entities(
+        [
+            SlackNumberEntity(
+                hass.data[DOMAIN][entry.entry_id][DATA_CLIENT],
+                NumberEntityDescription(
```

Comment:
I think this should be moved to a constant above `async_setup_entry`
There is no need to recreate the dataclass if there are multiple config entries

Reply:
This causes a typing issue with black reformatting it to a tuple.  ```homeassistant/components/slack/number.py:35: error: Argument 2 to "SlackNumberEntity" has incompatible type "Tuple[NumberEntityDescription]"; expected "EntityDescription"  [arg-type]```